### PR TITLE
Zarr reader

### PR DESCRIPTION
**Pull request by jennyfolkesson**
This PR adds support for reading ome-zarr input data into preprocessing. I'll save the writing of files for the next PR, which can be combined with adding gunpowder for data loading.

I've tested the preprocessing on data supplied by Soorya, running both zarr and tiff formats. Preprocessing with zarr is significantly faster despite still writing all tiles as numpy. One thing to note is that channel index differs if you generate frames metadata from zarrs and tiffs: zarr will index in the order specified in metadata, whereas tiff will assign channel indices to channel names in alphabetic order. So now that we're moving ahead with zarr (which always has channel names in metadata) we should move away from specifying channel indices and instead specify channel names in configs. This will also make it clearer what your input and output channels are and should be a simple PR.

We are already writing a metadata file in the tile directory with parameters including coordinates to all tiles that should be used in training, so hopefully this will come in handy when converting to gunpowder. Basically we'll rewrite the input data that has been flatfield corrected and normalized again in zarr format and then feed them and the metadata containing valid tile locations into gunpowder.

The commit history is pretty complex. It stems from initially developing the zarr reading on data that contained all positions in one zarr store. I had to make a minor modification to the zarr reader from waveorder to read the single position zarr format that is now the standard. I'll check in with someone who's developing waveorder to see if I'm using the classes right.
Thanks.

### CODE REVIEW

**Comment by jennyfolkesson** [comment]
I should also add that I broke my environment in my power9 docker container pretty thoroughly when trying to use zarr. Those power9 docker images will become completely obsolete after the old tensorflow dependencies are gone anyway so we'll need update the way we use the power9 servers soon.

**Comment by Christianfoley** [comment]
I found some issues during testing. I've enumerated them at #174.

**Review by Christianfoley — APPROVED** [review]
Hi @jennyfolkesson, this looks good! I left some comments about documentation and method descriptions. For the sake of time I also didn't look very hard at the tests, but I'm going to trust they're doing their job as long as they're passing.

@ziw-liu and I are in the progress of fixing the script that creates the one-store-per-position zarr structure. Moving forward, that will not be the structure of our data, and I think we should not support it to enforce uniformity. Let's fix that in the gunpowder PR, after we have rewritten the script.

File: README.md [hunk 1]
```diff
@@ -31,30 +71,41 @@ python micro_dl/cli/inference_script.py --config <train config yml> --gpu <int>
...
+
 ```buildoutcfg
 http://<your server name (e.g. fry)>:<whatever port you mapped to when starting up docker>
 ```
+
 You will need to copy/paste the token generated in your Docker container.
 
 ### Data Format
 
+Input data should be in the format of single page tiff files. If you use zarr files, you can convert
```
[the remark this answers is not in the document]
Comment (mattersoflight) [reply README.md:106]: This looks like a carryover from the previous README. We won't directly support single-page TIFF in microDL 2.0, but move to `czbiohub/iohub`.  
@ziw-liu The code to parse single-page TIFF datasets can be moved from microDL to iohub.

File: requirements.txt [hunk 2]
```diff
@@ -15,3 +16,4 @@ scipy==1.2.1
 tensorflow==1.13.1
 testfixtures==6.7.0
 tqdm
+zarr
```
[the remark this answers is not in the document]
Comment (jennyfolkesson) [reply requirements.txt:19]: Ah yes, especially for tqdm which has buggy older versions.

**Pull request merged**